### PR TITLE
[Do not merge] Setup CI tasks on drone

### PR DESCRIPTION
**Pull request by smeruelo**
Once this is merged, blog docker image will be created, pushed to Quay, and deployed to kubernetes by our Drone server.

Travis will create the docker image as well, as a premature way of finding errors. But it won't push it or deploy it.

Only two commits are relevant in this PR:
  78664ab  Add drone job
￼  dcc9aac  Remove docker-push from travis job ￼…
The rest of them belong to https://github.com/src-d/blog/pull/132

Please don't merge until a migration plan is discussed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,4 @@ go:
 
 script:
   - make build
-
-after_success:
-  - make docker-push-ci
\ No newline at end of file
+  - make docker-build
```
Comment (rporres) [inline .travis.yml]: Why do we create the images in travis? We don't need it

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+pipeline:
+
+# staging
+
+  build_stg:
+    image: alpine:latest
```
Comment (rporres) [inline .drone.yml]: Better use a fixed version, isn't it? Caddy image uses 3.6 - https://github.com/abiosoft/caddy-docker/blob/master/Dockerfile

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    debug: true
+    wait: true
+    when:
+      branch: master
+
+
+# production
+
+  build_prod:
+    image: alpine:latest
```
Comment (rporres) [inline .drone.yml]: ditto

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    registry: quay.io
+    repo: quay.io/srcd/blog
+    secrets: [ docker_username, docker_password ]
+    tag: ${DRONE_TAG}
+    file: Dockerfile
+    when:
+      event: tag
+
+  helm_deploy_prod:
+    image: quay.io/ipedrazas/drone-helm
```
Comment (rporres) [inline .drone.yml]: I'm not sure that we want ipedrazas image... I'd rather fork his repo and create one for us as I don't know how stable that image is going to be.

Comment (smeruelo) [reply .drone.yml]: Well, it's the only helm plugin that is linked in [Drone's website](http://plugins.drone.io/), and the one that the community is using and contributing to. I'd say it's stable enough.

File: .travis.yml (hunk 1 above)
Comment (smeruelo) [reply .travis.yml]: I missunderstood and thought we wanted to build it as a test. I'll remove it.

File: .drone.yml (hunk 4 above)
Comment (rporres) [reply .drone.yml]: Fair enough

**Comment by smeruelo** [comment]
Added changes requested by @rporres

**Comment by smeruelo** [comment]
I have added a few changes, fixing a few things that I learnt with the landing project since I opened this PR:

- Use staging branch to deploy to staging environment, as we agreed (https://github.com/src-d/minutes/pull/59)
- Removed the deployment workflow from `README` file since, as we agreed in the same meeting, it'll be documented somewhere else.
- Separate build step for PRs, it's cleaner this way
- Use quotes around commit hash (this avoids errors when hash starts by zero)
- Change resource's names to new ones (static IPs, etc)

Please review, @rporres @dpordomingo

**Comment by smeruelo** [comment]
Added last changes regarding what we decided in [this morning's meeting](https://github.com/src-d/minutes/pull/60).

@rporres @dpordomingo PTAL at it and, if you agree, I'll apply it (and migrate to k8s) tomorrow early in the morning.

**Pull request merged**